### PR TITLE
feat: Port boolean context processing to all storages

### PR DESCRIPTION
**Pull request by lynnagara**
Add equivalent processing for the 3 boolean contexts - device.online,
device.simulator and device.charging - to the errors, transactions and
discover storages. The results are now equivalent to those from the
events storage.

### CODE REVIEW

File: tests/query/processors/test_bool_context.py [hunk 1]
```diff
@@ -85,6 +88,79 @@ def test_events_boolean_context() -> None:
...
+                                        None,
+                                        "indexOf",
+                                        (
+                                            Column(None, None, "contexts.key"),
+                                            Literal(None, "device.charging"),
+                                        ),
+                                    ),
+                                ),
+                            ),
+                            # Column(None, None, "device_charging"),
```
Comment (evanh) [inline tests/query/processors/test_bool_context.py:150]: ?

**Review by evanh — APPROVED** [review]
I would run the Sentry tests against this, because I have a feeling there are some tests for this behaviour that might start to fail. Otherwise looks good.

**Comment by lynnagara** [comment]
> I would run the Sentry tests against this, because I have a feeling there are some tests for this behaviour that might start to fail. Otherwise looks good.

I believe this condition is physically impossible so that is unlikely, but will rerun the tests

File: tests/query/processors/test_bool_context.py (hunk 1 above)
Comment (lynnagara) [reply tests/query/processors/test_bool_context.py]: oops, deleted

**Pull request merged**